students/JackieCao/Lesson03/mailroom.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env python3
+"""
+mailroom
+"""
+donors = [("Fred Jones",[100, 200, 300]),
+        ("Amy Shumer", [2000, 4000, 1000]),
+        ("Bean Shing", [20, 400]),
+        ("Ann Shaw", [2, 400]),
+        ("King May", [200, 400, 100]),
+        ]
+# get a list of donor names
+donor_names = []
+for i in donors:
+    donor_names.append(i[0])
+
+# print donor names
+def print_donors():
+    for i in donors:
+        print(i[0])
+
+
+def thank_you():
+# ask use for donor's name
+    while True:
+        name = input("Enter a donor's name " 
+                "(or 'list' to see all donors or 'menu' to exit)>")
+        if name == "list":
+            print_donors()
+        elif name == "menu":
+            return
+        else:
+            break
+
+# ask user for donation amount
+    dollar = input("Enter a donation amount (or 'menu' to exit)>")
+    if dollar == "menu":
+        return
+    dollar = float(dollar)
+
+# add new name and donation amount to the donors history
+    if name in donor_names:
```

Comment:
donor_names was built at the start, so once you add a donor, it will be out of sync.